server.js

```diff
@@ -0,0 +1,59 @@
+
+'use strict';
+
+const bodyParser = require('./lib/body-parser');
+const http = require('http');
+const url = require('url');
+const queryString = require('querystring');
+const cowsay = require('cowsay');
+const PORT = process.env.PORT || 3000;
+
+const server = module.exports = http.createServer(function(req, res){
+  console.log(req.url);
+  req.url = url.parse(req.url);
+  req.url.query = queryString.parse(req.url.query);
+
+  if(req.method === 'POST') {
+
+    if(req.url.pathname === '/'){
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.write('Hello World!');
+      res.end();
+    }
+
+    if(req.url.pathname === '/cowsay') {
+      bodyParser(req, function(err) {
+        if(err) console.error(err);
+        let message = cowsay.say({text: req.body.text});
+        console.log('console check ', req.body.text);
```

Comment:
Note: You are going to want to remove these console logs prior to pushing to github.